Minor improvements for pipeline debugging.

wpietri opened the pull request.

bkorycki reviewed (commented):
Good idea! I'm not familiar w/ standard practice but does it make sense for all threads for a certain segment to have the same name? Instead of like threadname-1, threadname-2, etc.

wpietri commented:
> Good idea! I'm not familiar w/ standard practice but does it make sense for all threads for a certain segment to have the same name? Instead of like threadname-1, threadname-2, etc.

Great point! Fixed.

I also cleaned up some issues that happen when a pipe segment has an error, which should hopefully make debugging easier.

bkorycki commented:
Great thanks!
